syntax = "proto3";
package backup_protocol;
import "google/protobuf/empty.proto";

service Master {
    rpc NotifyMaster(Notification) returns (google.protobuf.Empty);
}

service Mapper{
    rpc StartReducer(Notification) returns (google.protobuf.Empty);
}

service Reducer{
    rpc StartMapper(Notification) returns (google.protobuf.Empty);
}

message Notification {
    string path = 1;
}

message Response {
    Status response=1;
}

enum Status {
    SUCCESS = 0;
    FAIL = 1;
}